src/Context/UserInfo.tsx

```diff
@@ -0,0 +1,122 @@
+import { createContext, useContext, useReducer } from "react";
+
+// Define Type for UserContextProvider
+type UserProps = {
+  children: React.ReactNode;
+};
+
+//Define Car
+interface UICar {
+  id: number,
+  brand: string, 
+  color: string
+}
+
+// Define Ui Person Interface
+interface UIPerson {
+  name: string;
+  lastName: string;
+  isLoggedIn: boolean;
+  cars: {id: number, brand: string, color: string}[];
+  toggleLogin?: () => void;
+  addUserCar?: (data: UICar) => void;
+  deleteUserCar?: (data: UICar) => void;
+  updateUserCar?: (data: UICar) => void;
+}
+
+// Define Basic Actions Type "Generic"
+enum ActionKind{
+  logged = 'logged',
+  add = 'add',
+  update = 'update',
+  delete = 'delete'
+}
+
+type Action = {
+  type: ActionKind,
+  payload: { id: number, brand: string, color: string}
+}
+
+
+const carReducer = ( state: UIPerson, action: Action ) =>{
+  switch (action.type) {
+    case ActionKind.add:
+      return {
+        ...state,
+        cars: [ ...state.cars , { ...action.payload } ]
```

Comment:
You don't need to spread the object, you can just assign it to the last slot in the array.
```suggestion
        cars: [ ...state.cars , action.payload ]
```